packages/amplify-category-auth/index.js

```diff
@@ -120,28 +120,55 @@ async function externalAuthEnable(context, externalCategory, resourceName, requi
   };
 
   try {
-    authProps = removeDeprecatedProps(authProps);
+    authProps = await removeDeprecatedProps(authProps);
     await copyCfnTemplate(context, category, authProps, cfnFilename);
-    saveResourceParameters(context, provider, category, authProps.resourceName, authProps, ENV_SPECIFIC_PARAMS);
+    await saveResourceParameters(context, provider, category, authProps.resourceName, authProps, ENV_SPECIFIC_PARAMS);
+    const resourceDirPath = path.join(amplify.pathManager.getBackendDirPath(), '/auth/', authProps.resourceName, 'parameters.json');
+    const authParameters = await amplify.readJsonFile(resourceDirPath);
     if (!authExists) {
       const options = {
         service: 'Cognito',
         providerPlugin: 'awscloudformation',
       };
-      const resourceDirPath = path.join(amplify.pathManager.getBackendDirPath(), '/auth/', authProps.resourceName, 'parameters.json');
-      const authParameters = amplify.readJsonFile(resourceDirPath);
 
       if (authParameters.dependsOn) {
         options.dependsOn = authParameters.dependsOn;
       }
       await amplify.updateamplifyMetaAfterResourceAdd(category, authProps.resourceName, options);
     }
+
+    // Update Identity Pool dependency attributes on userpool groups
+    const allResources = context.amplify.getProjectMeta();
+    if(allResources.auth && allResources.auth.userPoolGroups) {
+      let attributes;
```

Comment:
Same as above, can't we share this logic as well and just have it once in the codebase?

Reply:
Yeah, as I was diving through the code there's a refactor which I'm planning to basically consolidate the enable/update/externalAuth functions since it has a bunch of repeated code. It'll be a part of another PR.